File: test_phi3_client.py
#!/usr/bin/env python3
"""
Тест клієнта для microsoft/phi-3-mini-4k-instruct
"""

import sys
import os
sys.path.append('/Users/dev/Documents/GitHub/ATLAS/client-module/python_client')

from client import GitHubModelsClient

def test_phi3_mini():
    """Тестуємо модель microsoft/phi-3-mini-4k-instruct"""
    
    print("🚀 Тестуємо модель microsoft/phi-3-mini-4k-instruct")
    print("=" * 50)
    
    # Створюємо клієнт (він автоматично підхопить .env)
    client = GitHubModelsClient(
        proxy_url="http://localhost:5101/v1"
    )
    
    # Тестове повідомлення українською
    test_message = "Розкажи коротко про штучний інтелект українською мовою. Максимум 3 речення."
    
    print(f"📝 Запит: {test_message}")
    print("-" * 50)
    
    try:
        # Викликаємо модель
        result = client.chat_completion(
            model="microsoft/phi-3-mini-4k-instruct",
            messages=[
                {"role": "user", "content": test_message}
            ],
            max_tokens=200,
            temperature=0.7
        )
        
        print(f"✅ Успіх!")
        print(f"🤖 Модель: {result.get('model', 'невідомо')}")
        print(f"💬 Відповідь: {result.get('content', 'немає відповіді')}")
        print(f"📊 Токени: {result.get('usage', {}).get('total_tokens', 'невідомо')}")
        
    except Exception as e:
        print(f"❌ Помилка: {e}")
        print(f"🔍 Тип помилки: {type(e).__name__}")

if __name__ == "__main__":
    test_phi3_mini()

File: test_phi3_client.js
#!/usr/bin/env node
/**
 * Тест клієнта для microsoft/phi-3-mini-4k-instruct - Node.js версія
 */

import GitHubModelsClient from './client-module/nodejs_client/client.js';

async function testPhi3Mini() {
    console.log("🚀 Тестуємо модель microsoft/phi-3-mini-4k-instruct");
    console.log("=".repeat(50));
    
    // Створюємо клієнт
    const client = new GitHubModelsClient({
        proxyURL: "http://localhost:5101/v1"
    });
    
    // Тестове повідомлення українською
    const testMessage = "Розкажи коротко про штучний інтелект українською мовою. Максимум 3 речення.";
    
    console.log(`📝 Запит: ${testMessage}`);
    console.log("-".repeat(50));
    
    try {
        // Викликаємо модель
        const result = await client.chatCompletion({
            model: "microsoft/phi-3-mini-4k-instruct",
            messages: [
                { role: "user", content: testMessage }
            ],
            maxTokens: 200,
            temperature: 0.7
        });
        
        console.log("📊 Повна відповідь:", JSON.stringify(result, null, 2));
        
        if (result.success) {
            console.log("✅ Успіх!");
            console.log(`🤖 Модель: ${result.model || 'невідомо'}`);
            console.log(`💬 Відповідь: ${result.content || 'немає відповіді'}`);
            console.log(`📊 Токени: ${result.usage?.totalTokens || 'невідомо'}`);
        } else {
            console.log("❌ Помилка в результаті:");
            console.log(`🔍 Помилка: ${result.error}`);
        }
        
    } catch (error) {
        console.log(`❌ Помилка виконання: ${error.message}`);
        console.log(`🔍 Тип помилки: ${error.constructor.name}`);
        
        // Додаткова інформація для діагностики
        if (error.response) {
            console.log(`📡 Статус HTTP: ${error.response.status}`);
            console.log(`📄 Відповідь сервера:`, error.response.data);
        }
    }
}

// Запускаємо тест
testPhi3Mini().catch(console.error);
